go/pkg/pwapi/api_team-accept-invite.go

```diff
@@ -36,15 +36,38 @@ func (svc *service) TeamAcceptInvite(ctx context.Context, in *TeamAcceptInvite_I
 		return nil, pwdb.GormToErrcode(err)
 	}
 
-	// create team member
+	// check if user already has a team in this season
+	var existingSeasonMembership pwdb.TeamMember
+	err = svc.db.
+		Model(pwdb.TeamMember{}).
+		Joins("JOIN team on team.id = team_member.team_id AND team.season_id = ? AND team.deletion_status = ?", teamInvite.Team.SeasonID, pwdb.DeletionStatus_Active).
+		Preload("Team").
+		Where(pwdb.TeamMember{UserID: userID}).
+		First(&existingSeasonMembership).
+		Error
+	if !pwdb.IsRecordNotFoundError(err) {
+		return nil, errcode.ErrAlreadyHasTeamForSeason.Wrap(err)
+	}
+
 	teamMember := &pwdb.TeamMember{
 		UserID: userID,
 		TeamID: teamInvite.TeamID,
 	}
-	err = svc.db.
-		Create(&teamMember).Error
+	orgaMember := &pwdb.OrganizationMember{
+		UserID:         userID,
+		OrganizationID: teamInvite.Team.OrganizationID,
+	}
+	err = svc.db.Transaction(func(tx *gorm.DB) error {
+		// create team member
+		err = tx.Create(&teamMember).Error
+		if err != nil {
+			return pwdb.GormToErrcode(err)
+		}
+		// create orga member
+		return tx.Create(&orgaMember).Error
```

Comment:
can you make the `if err != nil` here too?

this way we will have the `pwdb.GormToErrcode` wrapping the error